### PR TITLE
Pull in bitcoin 0.21.1 & set taproot activation parameters

### PR DESCRIPTION
**Pull request by gertjaap**
This PR pulls in upstream changes to bitcoin up to v0.21.1. It also sets the parameters for taproot activation specific to Vertcoin.

### CODE REVIEW

**Comment by metalicjames** [comment]
Can you squash e5a83b3, 09432f6 and 58b861c (and maybe also e47e29a)?

Also 439aa50, 48c776b and 9b34165.

File: src/net.h [hunk 1]
```diff
@@ -74,10 +74,6 @@ static const bool DEFAULT_UPNP = USE_UPNP;
 #else
 static const bool DEFAULT_UPNP = false;
 #endif
-/** The maximum number of entries in mapAskFor */
```
Comment (metalicjames) [inline src/net.h:78]: Why don't we need these?

File: src/validation.cpp [hunk 2]
```diff
@@ -4769,7 +4767,11 @@ void LoadExternalBlockFile(const CChainParams& chainparams, FILE* fileIn, FlatFi
                     while (range.first != range.second) {
                         std::multimap<uint256, FlatFilePos>::iterator it = range.first;
                         std::shared_ptr<CBlock> pblockrecursive = std::make_shared<CBlock>();
-                        const int nHeight = ::BlockIndex()[it->first]->nHeight;
+                        CBlockIndex* idx = LookupBlockIndex(it->first);
```
Comment (metalicjames) [inline src/validation.cpp:4770]: Is this a change from upstream or VTC specific?

File: src/wallet/wallet.h [hunk 3]
```diff
@@ -66,7 +66,7 @@ static const CAmount DEFAULT_FALLBACK_FEE = 100000;
 //! -discardfee default
 static const CAmount DEFAULT_DISCARD_FEE = 10000;
 //! -mintxfee default
-static const CAmount DEFAULT_TRANSACTION_MINFEE = 1000;
+static const CAmount DEFAULT_TRANSACTION_MINFEE = 100000;
```
Comment (metalicjames) [inline src/wallet/wallet.h:69]: This is 1000 presently on master (https://github.com/vertcoin-project/vertcoin-core/blob/master/src/wallet/wallet.h#L73). Why the change? I remember there were some issues with the fee values in previous releases.

**Comment by gertjaap** [comment]
> Can you squash [e5a83b3](https://github.com/vertcoin-project/vertcoin-core/commit/e5a83b3c143ec11c6d35b272a26cbc586ce45b10), [09432f6](https://github.com/vertcoin-project/vertcoin-core/commit/09432f63f40643afb46e8c0599588a0649830d04) and [58b861c](https://github.com/vertcoin-project/vertcoin-core/commit/58b861c128743c3e2509dd8bc36b45b72bbfbf0c) (and maybe also [e47e29a](https://github.com/vertcoin-project/vertcoin-core/commit/e47e29a370cb03abbf5b505089c8f29450d26d36))?
> 
> Also [439aa50](https://github.com/vertcoin-project/vertcoin-core/commit/439aa507fc84dc1dbd507e31abade8be43ee3c55), [48c776b](https://github.com/vertcoin-project/vertcoin-core/commit/48c776b450c9eacec85acd3278687d0c2577d093) and [9b34165](https://github.com/vertcoin-project/vertcoin-core/commit/9b3416513442a6988564d7dbb8ec31a156664c69).

Squashed 09432f6, 58b861c and e47e29a. I can't squash additional changes/fixes with the merge commit. I also think we don't want that since it would obfuscate what was upstream and what was locally added.

Also squashed 439aa50, 48c776b and 9b34165.

File: src/validation.cpp (hunk 2 above)
Comment (gertjaap) [reply src/validation.cpp:4770]: `::BlockIndex()` was refactored out upstream

File: src/net.h (hunk 1 above)
Comment (gertjaap) [reply src/net.h:78]: This is removed upstream here: https://github.com/bitcoin/bitcoin/commit/1cff3d6cb017aea87d16cbda0768bbab256d16da

Comment (gertjaap) [reply src/net.h:78]: This is removed upstream here: https://github.com/bitcoin/bitcoin/commit/1cff3d6cb017aea87d16cbda0768bbab256d16da

File: src/wallet/wallet.h (hunk 3 above)
Comment (gertjaap) [reply src/wallet/wallet.h:69]: Thanks for catching this. I looked at the fixes you did [here](https://github.com/vertcoin-project/vertcoin-core/commit/939e9f2d8d50383b27bef47f21f60a4ffe395252) and [here](https://github.com/vertcoin-project/vertcoin-core/commit/c080428f37eb06bd77c9cc4c1846343034a28597) to ensure they weren't reverted by the merge of upstream changes. But I mixed up two variables. It seems that these parameters are still preserved they way they are in 0.17.1. Reverted my change

**Review by metalicjames — APPROVED** [review]

**Comment by metalicjames** [comment]
Sorry, I think I wasn't clear.

Can you make 09432f6, 58b861c and e47e29a one commit (part of the merge) and 439aa50, 48c776b and 9b34165 a subsequent commit (defining the new taproot params)?

**Comment by gertjaap** [comment]
It seems that `DEFAULT_TRANSACTION_MINFEE` does need to be 100k because there's a static assertion added that fails if that remains 1000:

https://github.com/vertcoin-project/vertcoin-core/pull/159/files#diff-157f588a09ff1da05b8c74acfcfb4da21917b6a97ed5d2e702228d621d23e66dR33
